src/main/scala/jio/package.scala

```diff
@@ -74,7 +64,37 @@ package object jio extends JioFiles with DecorateAsScala with DecorateAsJava {
     }
   }
 
-  trait Pathish[Rep] {
+  trait Metadataish {
+    def exists: Boolean
+    def path: Path
+    def allBytes: Array[Byte]
+    def nodeType: fs.NodeType
+    def size: Long
+    def permissions: PosixFilePermissions
+    def atime: Long
+    def mtime: Long
+
+    def name : String    = path.getFileName.to_s
+    def blockCount: Long = (size + blockSize - 1) / blockSize
+    def blockSize: Long  = 512 // FIXME
+  }
+
+  // this could have been an apply method in an object allowing the use of Metadata instead of Metadataish
+  // I however could not get myself to repeat all those arguments
```

Comment:
`getData: () => Array[Byte]` is very hacky and breaks the metadata abstraction. Metadata is decoupled from data. Yes, there is a way from Metadata to the corresponding data, but that path is of fundamental importance and has to be transparent and explicit. You can't bury it in `() => Array[Byte]` for half a dozen reasons... the data might be nonlocal, or sparse, or infinite, or not fit into memory, or may not even exist.

The way from the metadata to the data is through the filesystem. It's one of the two fundamental filesystem maps. One is translating paths into metadata, including a primary key which uniquely identifies the data - the inum. The other is translating inums (which are decoupled from paths) into data.